src/bernstein.jl

```diff
@@ -0,0 +1,79 @@
+export BernsteinBasis
+
+struct BernsteinBasis{N} <: AbstractP4MLBasis where {N} end
+
+BernsteinBasis(N::Integer) = BernsteinBasis{N}()
+
+Base.length(basis::BernsteinBasis{N}) where {N} = N
+
+natural_indices(basis::BernsteinBasis) = 0:length(basis)-1
```

Comment:
I've just changed the convention for this to 
```
[ (n = n,) for n = 0:length(basis)-1 ]
``` 
There are several reasons why this is important. It was also wrong in the code you started with.